examples/api/main.c

```diff
@@ -173,26 +174,26 @@ bool random_bit() { return 1 & rand(); }
 
 void fill_random_ballot(uint8_t *selections)
 {
-    uint8_t selected = 0;
+    uint32_t selected_count = 0;
     for (uint32_t i = 0; i < NUM_SELECTIONS; i++)
     {
-        if (!selected)
+        if (random_bit())
         {
-            selections[i] = random_bit() ? 1 : 0;
+            selections[i] = 1;
+            selected_count++;
         }
         else
         {
             selections[i] = 0;
         }
-        if (selections[i])
-        {
-            selected = 1;
-        }
-    }
-    if (!selected)
-    {
-        selections[NUM_SELECTIONS - 1] = 1;
     }
+
+    // ensure we dont have all trues or all falses
+    if (selected_count == NUM_SELECTIONS)
+        selections[0] = 0;
+    else if (selected_count == 0)
+        selections[0] = 1;
+
```

Comment:
The way things are set up right now, there is an exact number that must be selected for each contest. If fewer are selected, we use dummy votes to bring it up to the maximum, if more are selected it is an overvote and it should not be possible to create a valid encryption with proof.

Right now this work is on the client (this file, or the bmd) but it could be baked into the encrypt function, now that the code is moving towards a more general representation of ballots.

Reply:
I see... so it sounds like we shouldn't be calculating the `selected_count` then in `Validate_selections`, but instead store the calculated expected total selections as part of the api_config and pass that in during the call to API_EncryptBallot (like we currently do for num_selections).